charmcraft/commands/build.py

```diff
@@ -131,19 +127,28 @@ def handle_dispatcher(self, linked_entrypoint):
                 fileno = fh.fileno()
                 os.fchmod(fileno, os.fstat(fileno).st_mode | stat.S_IXUSR)
 
-        # bunch of symlinks, to support old juju
+        # bunch of symlinks, to support old juju; whatever is in the charm's hooks directory
+        # is respect, but also the mandatory ones are created if missing
         current_hookpath = self.charmdir / 'hooks'
         dest_hookpath = self.buildpath / 'hooks'
         dest_hookpath.mkdir()
-        for hookname in HOOK_NAMES:
-            current_hook = current_hookpath / hookname
+        if current_hookpath.exists():
+            current_hooks = set(current_hookpath.iterdir())
```

Comment:
I'm curious why you use a set here instead of a list.
Given you have to switch to a set later because it is just x.name you want to keep.

Reply:
Originally I was using them directly, then I found that I had to get the `.name` of each :/.

Moving them to a list for simplicity.